### PR TITLE
refactor: update `blas/ext/base/dfill` to follow current project conventions

### LINKED ISSUE #1434: [RFC]: refactor `blas/ext/base/dfill` to follow current project conventions
### Description

This RFC proposes refactoring [`@stdlib/blas/ext/base/dfill`](https://github.com/stdlib-js/stdlib/tree/1096686e16dd4515a7aaa5bdc35ff5803ec0349f/lib/node_modules/%40stdlib/blas/ext/base/dfill) to follow current project conventions.

These conventions are outlined in more detail in https://github.com/stdlib-js/stdlib/issues/788. Namely, we want to migrate from C++ add-on interfaces to C add-on interfaces and then make various style and simplification changes. 

Some example commits which perform most, if not all, of the requested changes include https://github.com/stdlib-js/stdlib/commit/d66d80cfcbab1da95cb5ae8cfae4a1fc4c5129d1 (saxpy),  https://github.com/stdlib-js/stdlib/commit/3c82fe7b6c0f9fd6ef02c86d6e2024d708ac8c14 (cswap), and https://github.com/stdlib-js/stdlib/commit/5b15c9675be434c0e11b1ec1e1f78d7b089abbcd (dasum).

### Related Issues

Related issues #788, #1152.

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by Snehil-Shah**
Resolves #1434

## Description

> What is the purpose of this pull request?

This pull request:

-   Refactors `blas/ext/base/dfill` to follow current project conventions

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #1434
-   progresses #1152

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

**Comment by Snehil-Shah** [comment]
@kgryte should I also update the license year or do they represent the year of first publication?

**Comment by kgryte** [comment]
Only in new files. No need to update the license year for existing files.

File: lib/node_modules/@stdlib/blas/ext/base/dfill/docs/repl.txt [hunk 1]
```diff
@@ -28,7 +28,7 @@
     Returns
     -------
     x: Float64Array
-        Input array `x`.
+        Output array.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/dfill/docs/repl.txt]:
```suggestion
        Input array.
```

We can actually leave as is, for this package, as the input array is what is returned. In other packages, we support providing an output array, but the intent of this package is to mutate the input array.

File: lib/node_modules/@stdlib/blas/ext/base/dfill/docs/repl.txt [hunk 2]
```diff
@@ -80,7 +79,7 @@
...
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/dfill/docs/repl.txt]:
```suggestion
        Input array.
```

File: lib/node_modules/@stdlib/blas/ext/base/dfill/docs/types/index.d.ts [hunk 3]
```diff
@@ -29,7 +29,7 @@ interface Routine {
 	* @param alpha - constant
 	* @param x - input array
 	* @param stride - stride length
-	* @returns `x`
+	* @returns output array
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/dfill/docs/types/index.d.ts]:
```suggestion
	* @returns input array
```

File: lib/node_modules/@stdlib/blas/ext/base/dfill/docs/types/index.d.ts [hunk 4]
```diff
@@ -49,7 +49,7 @@ interface Routine {
 	* @param x - input array
 	* @param stride - stride length
 	* @param offset - starting index
-	* @returns `x`
+	* @returns output array
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/dfill/docs/types/index.d.ts]:
```suggestion
	* @returns input array
```

File: lib/node_modules/@stdlib/blas/ext/base/dfill/docs/types/index.d.ts [hunk 5]
```diff
@@ -69,7 +69,7 @@ interface Routine {
 * @param alpha - constant
 * @param x - input array
 * @param stride - stride length
-* @returns `x`
+* @returns output array
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/dfill/docs/types/index.d.ts]:
```suggestion
* @returns input array
```

File: lib/node_modules/@stdlib/blas/ext/base/dfill/lib/dfill.js [hunk 6]
```diff
@@ -32,7 +32,7 @@ var M = 8;
 * @param {number} alpha - scalar
 * @param {Float64Array} x - input array
 * @param {integer} stride - index increment
-* @returns {Float64Array} input array
+* @returns {Float64Array} output array
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/dfill/lib/dfill.js]:
```suggestion
* @returns {Float64Array} input array
```

File: lib/node_modules/@stdlib/blas/ext/base/dfill/lib/dfill.native.js [hunk 7]
```diff
@@ -32,7 +32,7 @@ var addon = require( './../src/addon.node' );
 * @param {number} alpha - scalar
 * @param {Float64Array} x - input array
 * @param {integer} stride - index increment
-* @returns {Float64Array} input array
+* @returns {Float64Array} output array
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/dfill/lib/dfill.native.js]:
```suggestion
* @returns {Float64Array} input array
```

File: lib/node_modules/@stdlib/blas/ext/base/dfill/lib/ndarray.js [hunk 8]
```diff
@@ -33,7 +33,7 @@ var M = 8;
 * @param {Float64Array} x - input array
 * @param {integer} stride - index increment
 * @param {NonNegativeInteger} offset - starting index
-* @returns {Float64Array} input array
+* @returns {Float64Array} output array
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/dfill/lib/ndarray.js]:
```suggestion
* @returns {Float64Array} input array
```

File: lib/node_modules/@stdlib/blas/ext/base/dfill/lib/ndarray.native.js [hunk 9]
```diff
@@ -34,7 +35,7 @@ var addon = require( './dfill.native.js' );
 * @param {Float64Array} x - input array
 * @param {integer} stride - index increment
 * @param {NonNegativeInteger} offset - starting index
-* @returns {Float64Array} input array
+* @returns {Float64Array} output array
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/dfill/lib/ndarray.native.js]:
```suggestion
* @returns {Float64Array} input array
```

File: lib/node_modules/@stdlib/blas/ext/base/dfill/manifest.json [hunk 10]
```diff
@@ -34,7 +34,13 @@
 				"-lm"
 			],
 			"libpath": [],
-			"dependencies": []
+			"dependencies": [
+        "@stdlib/napi/export",
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/dfill/manifest.json:38]: @Snehil-Shah Would you mind converting this file to 2-space indentation? For ref, see https://github.com/stdlib-js/stdlib/blob/develop/.editorconfig#L147.

**Review by kgryte — CHANGES_REQUESTED** [review]
Thanks for working on this @Snehil-Shah. My only remaining comment is updating the indentation in the `manifest.json` file. Once resolved, this should be ready for merge.

File: lib/node_modules/@stdlib/blas/ext/base/dfill/manifest.json (hunk 10 above)
Comment (Snehil-Shah) [reply lib/node_modules/@stdlib/blas/ext/base/dfill/manifest.json:38]: @kgryte I think I'm missing something but the file already seems to be in 2-space indentation?

Comment (Snehil-Shah) [reply lib/node_modules/@stdlib/blas/ext/base/dfill/manifest.json:38]: @kgryte tried to fix it. is it okay now?

Comment (kgryte) [reply lib/node_modules/@stdlib/blas/ext/base/dfill/manifest.json:38]: It appears to be TAB indentation, not space indentation.

Comment (kgryte) [reply lib/node_modules/@stdlib/blas/ext/base/dfill/manifest.json:38]: Your changes are spaced indented, but the rest of the file is tab indented.

Comment (Snehil-Shah) [reply lib/node_modules/@stdlib/blas/ext/base/dfill/manifest.json:38]: @kgryte I think its fixed now

Comment (kgryte) [reply lib/node_modules/@stdlib/blas/ext/base/dfill/manifest.json:38]: Yes, it is. Thanks!

**Review by kgryte — APPROVED** [review]
LGTM. Thanks, @Snehil-Shah!

**Pull request merged**